xray/default_streaming_strategy.go

```diff
@@ -64,12 +66,15 @@ func (dSS *DefaultStreamingStrategy) StreamCompletedSubsegments(seg *Segment) []
 		seg.Subsegments[len(seg.Subsegments)-1] = nil
 		seg.Subsegments = seg.Subsegments[:len(seg.Subsegments)-1]
 
-		seg.ParentSegment.totalSubSegments--
+		atomic.AddUint32(&seg.ParentSegment.totalSubSegments, ^uint32(0))
 
 		// Add extra information into child subsegment
 		child.Lock()
 		child.beforeEmitSubsegment(seg)
-		cb, _ := json.Marshal(child)
+		cb, err:= json.Marshal(child)
+		if err!= nil{
```

Comment:
same here, if cb is nil, what will happen? We cannot just simply throw an error message here since customer might start spending time debugging this which should be their problem.

Reply:
Same as previous reply, if cb is nil - we don't stream the Subsegment. No error/panic happens. We shouldn't panic everywhere for possible marshaling error. IMHO, logging is the right strategy.